File: 05_hash_tables/scala/02_check_voter.scala
import scala.collection.mutable
val voted = mutable.Map[String, Boolean]()
def check_voter(name: String): Unit = {
  if(voted.exists(_ == (name,true))){
    println("kick them out!")
  } else{
    voted += (name -> true)
  }
}
check_voter("tom")
check_voter("mike")
check_voter("mike")

File: 05_hash_tables/golang/02_check_voter.go
package main

import "fmt"

var voted map[string]bool

func main() {
	voted = make(map[string]bool)
	check_voter("tom")
	check_voter("mike")
	check_voter("mike")
}

func check_voter(name string) {
	if voted[name] {
		fmt.Println("kick tem out!")
	} else {
		voted[name] = true
		fmt.Println("let tem vote!")
	}
}
